arcadia/aregion.cpp

```diff
@@ -1949,11 +1949,14 @@ int ARegion::CanTax(Unit *u)
 		Object *obj = (Object *) elem;
 		forlist ((&obj->units)) {
 			Unit *u2 = (Unit *) elem;
-			if (u2->guard == GUARD_GUARD && u2->IsReallyAlive())
-			    if(u2->type == U_GUARD || u2->type == U_GUARDMAGE) {
-			        if(town) return 0; //guards prevent taxing in towns
-			        if(u2->faction->ethnicity != u->faction->ethnicity) return 0; //guards prevent other ethnicities taxing
-				} else if (u2->GetAttitude(this, u) <= A_NEUTRAL) return 0;
+			if (u2->guard == GUARD_GUARD && u2->IsReallyAlive()) {
+				if (u2->type == U_GUARD || u2->type == U_GUARDMAGE) {
+					if (town) return 0; //guards prevent taxing in towns
+					if (u2->faction->ethnicity != u->faction->ethnicity) return 0; //guards prevent other ethnicities taxing
+				} else {
+					if (u2->GetAttitude(this, u) <= A_NEUTRAL) return 0;
+				}
+			}
 		}
```

Comment:
following the given code:
1. check for living guard (line 1952)
2. check if guarding unit is an actual guard (line 1953)
3. is this region a town? -> guards + town => can not tax (return 0/false) (line1954)
4. guard are NOT the same ethnicity than wannabe taxer -> can not tax (return 0/false) (line 1955)
5. otherwise: are guards unfriendly? -> can not tax (line 1956)
   (otherwise related to 1 or to 1 AND 2?)
6. can tax (after for-loop)

Prosposed solution: 
- consider lines 1953-1955 to be additions of arcadia
- therefore match the `else` with the `if` on line 1952
- that also preserves the logic of standard engine.